Network/Wai/Middleware/Static.hs

```diff
@@ -248,9 +253,19 @@ unsafeStaticPolicyWithOptions options p app req callback =
                                 if checkNotModified fileMeta (readHeader "If-Modified-Since") (readHeader "If-None-Match")
                                 then sendNotModified fileMeta strategy
                                 else sendFile fp (computeHeaders fileMeta strategy)
-                  else app req callback)
+                  else serveUpstream
+
+      mCandidateFile :: Maybe FilePath
+      mCandidateFile =
+          guard isHeadOrGet >>
           (tryPolicy p $ T.unpack $ T.intercalate "/" $ pathInfo req)
-    where
+          where
+            method :: Method
+            method = requestMethod req
+
+            isHeadOrGet :: Bool
+            isHeadOrGet = method == methodHead || method == methodGet
```

Comment:
Possibly dumb question: is there a reference which explains why `HEAD` and `GET` requests are the only reasonable requests for serving static content? I'd be willing to believe this, but it might be nice to point to something which codifies this wisdom.

Reply:
I'm not sure if there is a reference that specifically answers that question.  Personally, in the context of HTTP, I would just take this as give.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods lists all the methods with their purpose and semantic behavior and classifies them as "safe", "idempotent" and "cacheable".   That's the closest I could find.

Some notes:
- Apparently, `warp` doesn't send a body for `HEAD`, so I think we don't need to handle it specifically.
- As mentioned already, handling 405s from upstream in a 100% standard compliant way would need some more work.  The same is true for `OPTIONS`.  I don't think it is very critical to address this now, as I don't think that HTTP method discovery is commonly done.  Moreover, `OPTIONS` was broken before, so in the worst case (depending on what upstream does) it's still broken now.

Reply:
OK. I did at least find [another example of static middleware](https://github.com/senchalabs/connect/blob/83514d81f7f6e9f304fbb355d994c813660b7d2b/lib/middleware/static.js#L55) that only serves on `GET` or `HEAD` requests, so I'm comfortable going with this approach. I was mainly asking in case someone complains about their `wai-middleware-static` breaking upon the next release, in which case we could point them to a standard reference saying that that is a Very Bad Idea™. Oh well, I guess I can live without that :)